app-admin/systemrescuecd-x86/files/systemrescuecd.grub

```diff
@@ -0,0 +1,104 @@
+#!/bin/sh
+
+. /usr/share/grub/grub-mkconfig_lib
+
+if [ -r /etc/default/systemrescuecd ] ; then
+  . /etc/default/systemrescuecd
+fi
+
+# srcd = path to newest link to iso of systemrescuecd, created by ebuild
+srcd="/usr/share/systemrescuecd/systemrescuecd-x86-newest.iso"
+
+longname="SystemRescueCD"
+
+if [ -f "${srcd}" ]; then
+
+	# get version from real iso
+	version=`readlink ${srcd} | awk -F- '{print $3}' | sed 's/.iso//'`
```

Comment:
Instead of awk&sed, I suggest playing with variable substitution:

    version=$(readlink "${srcd}") # note that you had missing quotes
    version=${version#systemrescuecd-x86-}
    version=${version%.iso}

This is faster and more portable ;-).